wp-content/civi-extensions/civiglific/api/v3/Civiglific/CivicrmGlificContactSyncCron.php

```diff
@@ -0,0 +1,459 @@
+<?php
+
+/**
+ * @file
+ */
+
+use Civi\Api4\Contact;
+use Civi\Api4\SubscriptionHistory;
+use CRM\Civiglific\GlificHelper;
+use Civi\Api4\Phone;
+use Civi\Api4\GroupContact;
+use Civi\Api4\GlificGroupMap;
+use GuzzleHttp\Client;
+
+/**
+ * Define API spec.
+ */
+function _civicrm_api3_civiglific_civicrm_glific_contact_sync_cron_spec(&$spec) {}
+
+/**
+ * Cron job to sync contacts between CiviCRM group and Glific group.
+ */
+function civicrm_api3_civiglific_civicrm_glific_contact_sync_cron($params) {
+  $returnValues = [];
+
+  // Batch size to process contacts in smaller chunks to avoid timeout.
+  $batchSize = 50;
+  // Sleep time between batches (seconds) to reduce API rate limits and server load.
```

Comment:
This is comment is useful because it explains  the **“why”** something is done, not **“what.”**